src/Prairie/Advanced.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+module Prairie.Advanced
+    where
+
+import Data.Functor.Const (Const(..))
+import Data.Functor.Identity (Identity(..))
+import Data.Text (Text)
+import Prairie.Class (Field, getRecordField, recordFieldLabel, tabulateRecordA)
+import qualified Prairie.Class as Simple
+import Prelude hiding (sequenceA, zipWith)
+
+newtype Record f rec = Record
```

Comment:
This name clashes with the `Record` class - can you pick a different name that suggests what is different about it?

I would be open to changing the *class* to `IsRecord` or something, but `Record` feels a little overly-generic of a name for what this is/does

Reply:
It's an Advanced.Record so possibly just ARecord. Alternatively HigherKindedRecord .

Reply:
This form actually came up in the Mercury slack! It was named `Decorated` there, though I'm not attached to it.

I am definitely inclined to say that we want this in there, though I think the naming/structure could use some work. `Advanced` feels a bit of a misnomer since the whole library is kind of "advanced".

Reply:
What about `Wrap`?

- `Record` -> `Wrap`
- `toAdvanced` -> `wrap`
- `fromAdvanced` -> `unwrap`

Reply:
To be clear I stole the term advanced from the way large-anon uses it. Decorated makes some amount of sense. Another possible option is Distributed since we are distributing f to every field